interface-definitions/interfaces-tinc.xml.in

```diff
@@ -0,0 +1,476 @@
+                    <description>Use a dummy interface. No packets are ever read or written to a virtual network device. Useful for testing, or when setting up a node that only forwards packets for other nodes. </description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>raw_socket</format>
+                    <description>Open a raw socket, and bind it to a pre-existing Interface (eth0 by default). All packets are read from this interface. Packets received for the local node are written to the raw socket. However, at least on Linux, the operating system does not process IP packets destined for the local host</description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>multicast</format>
+                    <description>Open a multicast UDP socket and bind it to the address and port (separated by spaces) and optionally a TTL value specified using Device. Packets are read from and written to this multicast socket. This can be used to connect to UML, QEMU or KVM instances listening on the same multicast address. Do NOT connect multiple tinc daemons to the same multicast address, this will very likely cause routing loops. Also note that this can cause decrypted VPN packets to be sent out on a real network if misconfigured </description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>tun</format>
+                    <description>Set type to tun. Depending on the platform, this can either be with or without an address family header (see below) </description>
+                  </valueHelp>
+                  <constraint>
+                    <regex>(dummy|raw_socket|multicast|tun)</regex>
+                  </constraint>
+                </properties>
+                <defaultValue>tap</defaultValue>
+              </leafNode>
+              <leafNode name="mode">
+                <properties>
+                  <help>Tinc VPN interface device mode</help>
+                  <completionHelp>
+                    <list>switch hub</list>
+                  </completionHelp>
+                  <valueHelp>
+                    <format>switch</format>
+                    <description>switch device</description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>hub</format>
+                    <description>hub device</description>
+                  </valueHelp>
+                  <constraint>
+                    <regex>(switch|hub)</regex>
+                  </constraint>
+                </properties>
+                <defaultValue>router</defaultValue>
+              </leafNode>
+            </children>
+          </node>
+          #include <include/interface-ipv4.xml.i>
+          #include <include/interface-ipv6.xml.i>
+          #include <include/interface-disable.xml.i>
+          #include <include/interface-mac.xml.i>
+          <leafNode name="broadcast-type">
```

Comment:
I feel that the default value should be printed for the customer to choose, and whether the customer chooses this value or not, it represents a setting. When the user does not choose, we use the default value, but if the user chooses, the user may have the right to know the default setting What is it, and can choose it clearly

Reply:
There is a consensus when creating options for vyos that the default value should not be a setting in the cli. this means that for the user to use the default value he must delete the custom setting. the type of the value could be documented in a help-string on the main option. this have been discussed multiple times for mulitple options prior to this.

Reply:
the consensus goes somthing like this: "there should not be two options that correspond to the same output value. eg. the value of not defining the option should not have a corresponding output value as a defined option"